pkg/R/sparkR.R

```diff
@@ -121,45 +119,46 @@ sparkR.init <- function(
   if (sparkRExistingPort != "") {
     sparkRBackendPort <- sparkRExistingPort
   } else {
+    path <- tempfile(pattern = "backend_port")
     if (Sys.getenv("SPARKR_USE_SPARK_SUBMIT", "") == "") {
       launchBackend(classPath = cp,
                     mainClass = "edu.berkeley.cs.amplab.sparkr.SparkRBackend",
-                    args = as.character(sparkRBackendPort),
+                    args = path,
                     javaOpts = paste("-Xmx", sparkMem, sep = ""))
     } else {
       # TODO: We should deprecate sparkJars and ask users to add it to the
       # command line (using --jars) which is picked up by SparkSubmit
       launchBackendSparkSubmit(
           mainClass = "edu.berkeley.cs.amplab.sparkr.SparkRBackend",
-          args = as.character(sparkRBackendPort),
+          args = path,
           appJar = .sparkREnv$assemblyJarPath,
           sparkHome = sparkHome,
           sparkSubmitOpts = Sys.getenv("SPARKR_SUBMIT_ARGS", ""))
     }
+    for (i in 1:100) {
+      if (file.exists(path)) {
+        break
+      }
+      Sys.sleep(0.1)
```

Comment:
Yeah but there is some trade-off if we want to handle wait times up to 60 seconds (due to a slow machine) and keeping it responsive. Checking every 0.1 second would result in 600 system calls (`file.exists` needs to call `stat` I'd think) so we need to balance this out. 

Actually if 1 second wait granularity is fine, then we could just check every second 60 times. Do you think that is better than exponential back-off ?